native/Avalonia.Native/src/OSX/window.mm

```diff
@@ -1076,6 +1076,16 @@ @implementation AvnWindow
     bool _isAppMenuApplied;
 }
 
++(void)closeAll
+{
+    auto numWindows = [[NSApp windows] count];
```

Comment:
It's better to make a local copy of `[NSApp windows]` via https://developer.apple.com/documentation/foundation/nsarray/1412169-initwitharray?language=objc

Reply:
fixed